src/cares.c

```diff
@@ -454,8 +455,10 @@ query_txt_cb(void *arg, int status,int timeouts, unsigned char *answer_buf, int
 {
     PyGILState_STATE gstate = PyGILState_Ensure();
     int parse_status;
-    struct ares_txt_reply *txt_reply, *txt_ptr;
-    PyObject *dns_result, *errorno, *tmp, *result, *callback;
+    struct ares_txt_ext *txt_reply, *txt_ptr;
+    PyObject *dns_result, *errorno, *tmp_obj, *result, *callback;
+    long assembled_ttl;
```

Comment:
no need to keep the assembled TTL, just set it when you create the structsequence since it's the same for all chunks